tools/cli/adminDBCleanCommand.go

```diff
@@ -0,0 +1,309 @@
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package cli
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"math"
+	"os"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/urfave/cli"
+
+	"github.com/uber/cadence/common/log/loggerimpl"
+	"github.com/uber/cadence/common/persistence"
+	cassp "github.com/uber/cadence/common/persistence/cassandra"
+	"github.com/uber/cadence/common/quotas"
+)
+
+type (
+	// ShardCleanReport represents the result of cleaning a single shard
+	ShardCleanReport struct {
+		ShardID         int
+		TotalDBRequests int64
+		Handled         *ShardCleanReportHandled
+		Failure         *ShardCleanReportFailure
+	}
+
+	// ShardCleanReportHandled is the part of ShardCleanReport of executions which were read from corruption file
+	// and were attempted to be deleted
+	ShardCleanReportHandled struct {
+		TotalExecutionsCount     int64
+		SuccessfullyCleanedCount int64
+		FailedCleanedCount       int64
+	}
+
+	// ShardCleanReportFailure is the part of ShardCleanReport that indicates a failure to clean some or all
+	// of the executions found in corruption file
+	ShardCleanReportFailure struct {
+		Note    string
+		Details string
+	}
+
+	// CleanProgressReport represents the aggregate progress of the clean job.
+	// It is periodically printed to stdout
+	CleanProgressReport struct {
+		NumberOfShardsFinished     int
+		TotalExecutionsCount       int64
+		SuccessfullyCleanedCount   int64
+		FailedCleanedCount         int64
+		TotalDBRequests            int64
+		DatabaseRPS                float64
+		NumberOfShardCleanFailures int64
+		ShardsPerHour              float64
+		ExecutionsPerHour          float64
+	}
+
+	// CleanOutputDirectories are the directory paths for output of clean
+	CleanOutputDirectories struct {
+		ShardCleanReportDirectoryPath    string
+		SuccessfullyCleanedDirectoryPath string
+		FailedCleanedDirectoryPath       string
+	}
+
+	// ShardCleanOutputFiles are the files produced for a clean of a single shard
+	ShardCleanOutputFiles struct {
+		ShardCleanReportFile    *os.File
+		SuccessfullyCleanedFile *os.File
+		FailedCleanedFile       *os.File
+	}
+)
+
+// AdminDBClean is the command to clean up executions
+func AdminDBClean(c *cli.Context) {
+	lowerShardBound := c.Int(FlagLowerShardBound)
+	upperShardBound := c.Int(FlagUpperShardBound)
+	numShards := upperShardBound - lowerShardBound
```

Comment:
I don't quite understand this shard range. From my understanding, shardID is quite random for different workflows. It's unlikely the shard we want to clean up happens to fall in a small range. 

I can see two use cases here:
1. User want to clean up all the shards. In this case we should ask the user to specify the total number of shards.
2. User want to clean up some specific shard in question. In this case we should accept a list of shardIDs. 

Those two flags should be mutual exclusive, meaning only one of them can be specified at a time.

Reply:
The use case is this: scan runs over ranges and produces directories which contain different shards. Now lets say you want to do a a full clean. The way you will do this is give an input directory and a shard range. If you want to clean just one shard then you will do a scan of that one shard and then do a clean of that one shard by giving a range of size 1.

Reply:
I skip shards which don't have scan files as part of the clean up. Maybe if I specified this it would be more clear? Thoughts?